internal/database/repository/queue_int_test.go

```diff
@@ -495,6 +495,203 @@ func TestAddQueues_Integration(t *testing.T) {
 
 }
 
+func TestUpdateQueue_Integration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode")
+	}
+
+	ctx := context.Background()
+	connPool := database.NewTestConnectionPool(ctx, t)
+
+	repo, err := NewPostgresRepository(connPool)
+	if err != nil {
+		t.Fatalf("could not create repository: %v", err)
+	}
+
+	tests := []struct {
+		name           string
+		existingQueues []*dao.PartitionQueueDAOInfo
+		queueToUpdate  *dao.PartitionQueueDAOInfo
+		expectedError  error
+	}{
+		{
+			name: "Update root queue when root queue exists",
+			existingQueues: []*dao.PartitionQueueDAOInfo{
+				{
+					Partition:       "default",
+					QueueName:       "root",
+					CurrentPriority: 0,
+				},
+			},
+			queueToUpdate: &dao.PartitionQueueDAOInfo{
+				Partition:       "default",
+				QueueName:       "root",
+				CurrentPriority: 1,
+			},
+			expectedError: nil,
+		},
+		{
+			name:           "Update root queue when root queue does not exist",
+			existingQueues: nil,
+			queueToUpdate: &dao.PartitionQueueDAOInfo{
+				Partition:       "default",
+				QueueName:       "root",
+				CurrentPriority: 1,
+			},
+			expectedError: fmt.Errorf("queue not found: %s", "root"),
```

Comment:
Usually, you shouldn't test for the error message itself, but rather on type of error or existence of it. If someone changes the error format to `queue %q not found`, then this test will fail, while there is nothing wrong with it. If I'm testing for existence of an error, I usually go with `expectedErr bool`. Then, you basically `require.Equal(t, expectedErr, err != nil)`